scanners/ios/icon/icon.go

```diff
@@ -0,0 +1,142 @@
+package icon
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/bitrise-core/bitrise-init/models"
+	"github.com/bitrise-core/bitrise-init/utility"
+	"github.com/bitrise-io/go-utils/log"
+)
+
+// LookupPossibleMatches returns possible ios app icons,
+// in a map with key of a id (sha256 hash converted to string), value of icon path
+func LookupPossibleMatches(searchPath string, basepath string) (models.Icons, error) {
```

Comment:
It seems the given target's `buildPhases` contains which .xcassets will be copied into the .app bundle (a `PBXResourcesBuildPhase` entry, which `files` node contains .xcassets).
The target's `ASSETCATALOG_COMPILER_APPICON_NAME ` describes which asset is the app icon from the copied `.xcassets`.

The strange thing is that multiple copied `.xcassets` can contain app icon assets with the same name, in this case Xcode is buggy a bit and you can not select which one you want to use (there is a dropdown in xcode and you can select which catalogs which assets you want to use, but you can not switch between app icon assets with the same name in different assets).